midend/lib/Conversion/MatMulOptimization/MatMulTransposeBVec.cpp

```diff
@@ -0,0 +1,161 @@
+        const Value step = 
+            rewriter.create<arith::ConstantIndexOp>(loc,veSize);
+        
+        const Value aRow = rewriter.create<memref::DimOp>(loc,A,c0);
+        const Value bRow = rewriter.create<memref::DimOp>(loc,B,c0);
+        const Value bCol = rewriter.create<memref::DimOp>(loc,B,c1);
+        
+        SmallVector<Value,8> lowerBounds(2,c0);
+        SmallVector<Value,8> uperBounds {aRow,bRow/*bCol*/};
+        SmallVector<int64_t,8> steps{1,1};
+        
+        affine::buildAffineLoopNest(
+            rewriter,loc,lowerBounds,uperBounds,steps,
+            [&](OpBuilder &builder,Location loc,ValueRange ivs){
+                Value sum_0 = buddy::insertZeroConstantOp(ctx, rewriter, loc, eleTy);
+                // Create loop based on vector size.
+                auto lbMap = mlir::AffineMap::get(0, 0, ctx);
+                auto ubMap = mlir::AffineMap::get(0, 0, ctx);
+                auto sum= builder.create<affine::AffineForOp>(
+                loc,ValueRange{c0}, builder.getDimIdentityMap(),ValueRange{bCol}, builder.getDimIdentityMap()
+                    ,veSize,ValueRange{sum_0},
+                    [&](OpBuilder &nestedBuilder, Location nestedLoc, Value iv,
+                    ValueRange itrArgs) {
+                        Value aVec = builder.create<mlir::vector::LoadOp>(
+                            loc,vectorTy,A,ValueRange{ivs[0],iv}
+                        );
+                        Value bVec = builder.create<mlir::vector::LoadOp>(
+                            loc,vectorTy,B,ValueRange{ivs[1],iv}
+                        );
+                        Value resvec = builder.create<arith::MulFOp>(loc,aVec,bVec);
+                        Value ans = builder.create<mlir::vector::ReductionOp>(
+                            loc,mlir::vector::CombiningKind::ADD,resvec
+                        );
+                        Value sum = builder.create<arith::AddFOp>(loc,itrArgs[0],ans);
+                        builder.create<affine::AffineYieldOp>(loc,sum);
+                    }
+                );
+                builder.create<affine::AffineStoreOp>(loc,sum.getResult(0),C,ValueRange{ivs[0],ivs[1]});
+            }
+        );
+        // clang-format on
+        rewriter.eraseOp(op);
+        return success();   
+    }
+private:
+    int64_t veSize;
+};
+} // end anonymous namespace
+
+//===----------------------------------------------------------------------===//
+// MatMulVectorizationPass
+//===----------------------------------------------------------------------===//
+
+namespace{
+    class MatMulTransposeBVecPass
+        :public PassWrapper<MatMulTransposeBVecPass,OperationPass<ModuleOp>>{
+public:
+    MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(MatMulTransposeBVecPass)
+    StringRef getArgument() const final{ return "transpose_matmul_bvectorization"; }
+    StringRef getDescription() const final { return "MatMul Vectorization second version.MatMul receive tensortype oprands."; }
```

Comment:
"MatMul Vectorization second version.MatMul receive tensortype oprands." This is not a good description for this pass.